src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -485,4 +492,60 @@ export class SavedObjectsService
       ),
     });
   }
+
+  private selectRepository(config: SavedObjectConfig) {
+    const savedObjectsRepository = OpensearchSavedObjectsRepository;
+    switch (config.dbType) {
+      case 'opensearch':
+        return savedObjectsRepository;
+      case 'postgres':
+        this.createTable(config);
+        return PostgresSavedObjectsRepository;
+      default:
+        return savedObjectsRepository;
+    }
+  }
+
+  private async createTable(config: SavedObjectConfig) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const pg = require('pg');
+    const dbName = 'opensearch_dashboards';
+    const connUrl = `postgres://${config.dbUserName}:${config.dbPassword}@${config.dbHostName}:${config.dbPort}`;
+    postgresClient = new pg.Client(connUrl);
+    postgresClient.connect();
+
+    await postgresClient
+      .query(`CREATE DATABASE ${dbName}`)
+      .then((res: any) => {
+        this.logger.info(`DATABSE ${dbName} is successfully created`);
+      })
+      .catch((error: any) => {
+        this.logger.info(error);
+      });
+    postgresClient.end();
```

Comment:
Can you please share the ideas why we are using `end` command instead of `commit` to save changes here?

Reply:
Commit throws error :`TypeError: postgresClient.commit is not a function`. Anyhow this code block has bug and need to fix better way. I will raise another revision.